Locales/gap/ConstructibleObjectsAsUnionOfSingleDifferences.gi

```diff
@@ -151,16 +161,25 @@ InstallMethod( BooleanAlgebraOfConstructibleObjectsAsUnionOfDifferences,
     AddDirectProduct( C,
       function( cat, L )
         
-        return Iterated( L, { A, B } -> BinaryDirectProduct( cat, A, B ) );
+        if IsEmpty( L ) then
+            return TerminalObject( C );
+        else
+            return Iterated( L, { A, B } -> BinaryDirectProduct( cat, A, B ) );
+        fi;
```

Comment:
You can use `Iterated` with three arguments (part of CAP) to avoid the `if ... else ...`.

Reply:
Done, thank you very much :)